src/order/ideal.lean

```diff
@@ -299,14 +336,80 @@ end
 
 end semilattice_sup_ideal_inter_nonempty
 
-section semilattice_sup_bot
-variables [semilattice_sup_bot P]
+section ideal_Inter_nonempty
+
+variables [preorder P] [ideal_Inter_nonempty P]
 
 @[priority 100]
-instance semilattice_sup_bot.ideal_inter_nonempty : ideal_inter_nonempty P :=
-{ inter_nonempty := λ _ _, ⟨⊥, ⟨bot_mem, bot_mem⟩⟩ }
+instance ideal_Inter_nonempty.ideal_inter_nonempty : ideal_inter_nonempty P :=
+{ inter_nonempty := λ _ _, begin
+    cases ideal_Inter_nonempty.exists_all_mem ‹_› with a ha,
+    exact ⟨a, ha _, ha _⟩
+  end
+}
+
+variables {α β γ : Type*} {ι : Sort*}
+
+lemma ideal_Inter_nonempty.ideal_Inter_nonempty {f : ι → ideal P} :
+(⋂ x, (f x : set P)).nonempty :=
+begin
+  cases ideal_Inter_nonempty.exists_all_mem ‹_› with a ha,
+  exact ⟨a, by simp [ha]⟩
+end
+
+lemma ideal_Inter_nonempty.ideal_bInter_nonempty {f : α → ideal P} {s : set α} :
+(⋂ x ∈ s, (f x : set P)).nonempty :=
+begin
+  cases ideal_Inter_nonempty.exists_all_mem ‹_› with a ha,
+  exact ⟨a, by simp [ha]⟩
+end
+
+end ideal_Inter_nonempty
+
+section semilattice_sup_ideal_Inter_nonempty
+
+variables [semilattice_sup P] [ideal_Inter_nonempty P] {x : P} {I J K : ideal P}
+
+def Inf (s : set (ideal P)) : ideal P :=
+{ carrier := ⋂ (I ∈ s), (I : set P),
+  nonempty := ideal_Inter_nonempty.ideal_bInter_nonempty,
+  directed := λ x hx y hy, ⟨x ⊔ y, ⟨λ S ⟨I, hS⟩,
+    begin
+      simp [← hS],
+      assume hI,
+      rw set.mem_bInter_iff at *,
+      tauto,
+    end,
+    le_sup_left, le_sup_right⟩⟩,
+  mem_of_le := λ x y hxy hy,
+    begin
+      rw set.mem_bInter_iff at *,
+      exact λ I hI, mem_of_le I ‹_› (hy I hI)
+    end }
+
+instance : has_Inf (ideal P) :=
+{ Inf := ideal.Inf }
```

Comment:
I think you should inline the implementation directly into the `has_Inf`. If you're looking to make this shorter, perhaps extracted `directed` as a standalone lemma.